packages/near-api-js/src/contract.ts

```diff
@@ -98,14 +128,27 @@ export interface ContractMethods {
     abi: AbiRoot;
 }
 
+interface ViewContractOptions {
+    connection: Connection;
+    contractId: string;
+    viewContractMethods: ViewContractMethods;
+}
+
+interface FullContractOptions {
+    account: Account;
+    contractId: string;
+    contractMethods: ContractMethods;
+}
```

Comment:
Given what I've said about removing `Account` from any method/constructor signatures, I don't think there's a strong case for making a distinction between a view-only `Contract` and one that also has change methods. For now it's OK for there to be an implicit understanding that the `Connection` instance must have a `Signer` attached to it in order to sign transactions.

Later on we can introduce breaking changes that make this more explicit, e.g. by providing only the `JsonRpcProvider` rather than an entire `Connection` and by expanding the arguments to change methods to include the signing account ID and a way to sign the transaction.